core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -704,6 +741,284 @@ export class LoggedInUser extends BaseUser {
+    await this.type(reportTextArea, reportMessage);
+    await this.page.click('.e2e-test-submit-report-button');
+    await this.page.waitForSelector('.modal-content');
+    await this.page.click(
+      'oppia-exploration-successfully-flagged-modal button'
+    );
+    /** Giving explicit timeout because we need to wait for small
+     * transition to complete. We cannot wait for the next element to click
+     * using its selector as it is instantly loaded in the DOM but cannot
+     * be clicked until the transition is completed.
+     */
+    await this.page.waitForTimeout(500);
+    showMessage(`Exploration reported with message ${reportMessage}.`);
+  }
+
+  /**
+   * Function for giving feedback for an exploration.
+   */
+  async giveFeedbackToExploration(feedbackMessage: string): Promise<void> {
+    await this.page.click(explorationFeedbackButton);
+    await this.type(feedbackTextArea, feedbackMessage);
+    await this.page.click('.e2e-test-exploration-feedback-submit-btn');
+    await this.page.waitForSelector('ngb-popover-window');
+    showMessage(`Feedback has been given with the message: ${feedbackMessage}`);
+  }
+
+  /**
+   * This function checks whether explorations with the given title are present in the Play Later.
+   */
+  async expectExplorationWithTitleToBePresentInPlayLater(
+    title: string,
+    shouldBePresent: boolean
+  ): Promise<void> {
+    const allExplorations = await this.page.$$(playLaterExplorationCard);
+
+    if (!shouldBePresent) {
+      if (allExplorations.length > 0) {
+        throw new Error('There are some explorations present in Play Later');
+      } else {
+        showMessage(
+          `Exploration with title ${title} does not exists in Play Later`
+        );
+      }
+      return;
+    }
+
+    if (allExplorations.length === 0) {
+      throw new Error(
+        `Exploration with title ${title} does not exist in play later`
+      );
+    }
+
+    const explorationTitle = await allExplorations[0].$eval(
+      playLaterExplorationCardTitle,
+      element => (element as HTMLElement).innerText
+    );
+
+    if (explorationTitle === title) {
+      showMessage(`Exploration with title ${title} exists in Play Later`);
+    } else {
+      throw new Error(
+        `Exploration with title ${title} does not exist in play later`
+      );
+    }
+  }
+
+  /**
+   * This function to add exploration to play later.
+   */
+  async addExplorationToPlayLater(): Promise<void> {
+    await this.page.hover(communityExplorationCard);
+    await this.page.click('.e2e-test-add-to-playlist-btn');
+    await this.page.waitForSelector('.e2e-test-toast-message');
+  }
+
+  /**
+   * This function to remove exploration from play later.
+   */
+  async removeExplorationFromPlayLater(): Promise<void> {
+    await this.page.hover(playLaterExplorationCard);
+    await this.page.click('.remove-icon');
+    await this.page.waitForSelector(confirmRemoveButton);
+    await this.page.click(confirmRemoveButton);
```

Comment:
We already have utility methods that combine waiting before the user action in `this.clickOn`, `this.select`, `this.type`, etc. (you can find a list of these in `/core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts`). Please replace this and every other user action with the corresponding utility method.

Reply:
Done